packages/components/src/number-control/index.js

```diff
@@ -43,10 +43,13 @@ export function NumberControl(
 	const baseStep = isStepAny ? 1 : parseFloat( step );
 	const baseValue = roundClamp( 0, min, max, baseStep );
 	const constrainValue = ( value, stepOverride ) => {
+		let val = parseFloat( value );
+		val = isNaN( val ) ? 0 : val;
```

Comment:
Note that it's possible for `Infinity` to also be returned from `parseFloat`. It looks like we're clamping this properly, but it shows up in the UI:

https://user-images.githubusercontent.com/1270189/152067041-ead0e767-c8aa-4cea-8d80-b51937c0e4a3.mp4